cpp/include/cudf/types.h

```diff
@@ -85,6 +85,36 @@ typedef struct {
 	// here we can also hold info for decimal datatype or any other datatype that requires additional information
 } gdf_dtype_extra_info;
 
+/**---------------------------------------------------------------------------*
+ * @union gdf_data
+ * @brief Union used for scalar type.
+ * It stores a unique value for scalar type.
+ * It has a direct relationship with the gdf_dtype.
+ *---------------------------------------------------------------------------**/
+// TODO: #1119 Use traits to set `gdf_data` elements
+typedef union {
+  int8_t   si08;  /**< GDF_INT8      */
+  int16_t  si16;  /**< GDF_INT16     */
+  int32_t  si32;  /**< GDF_INT32     */
+  int64_t  si64;  /**< GDF_INT64     */
+  float    fp32;  /**< GDF_FLOAT32   */
+  double   fp64;  /**< GDF_FLOAT64   */
+  int32_t  dt32;  /**< GDF_DATE32    */
```

Comment:
Looks like we're missing an entry for `gdf_category`.

Reply:
binary ops aren't defined on `gdf_category`. How would you calculate `Zebra % Elephant`

Reply:
Ah, yeah, good point. I guess the only binary ops that make sense on a category column are   

```
GDF_EQUAL,          /**< operator == */
  GDF_NOT_EQUAL,      /**< operator != */
```

We can wait for someone to complain about not having those operations on categorical columns before worrying about it. No need to add `gdf_category` now.

Reply:
Wait. This doesn't sound right. A single column element can very well be a `gdf_category`. The issue @devavret  brought up doesn't mean it can be missing here. IMHO.